packages/flutter_tools/lib/src/xcode_project.dart

```diff
@@ -357,6 +369,33 @@ class IosProject extends XcodeBasedProject {
         }
       }
     }
+
+    // The build settings of the watchOS companion app's scheme
```

Comment:
Can you update the "The Info.plist file of a target contains the key WKCompanionAppBundleIdentifier" comment above to explain why the Info.plist is checked first, then with a fallback to the more expensive build setting check?

Reply:
Updated in https://github.com/flutter/flutter/pull/113956/commits/25df5b631ab3acaffc8d9bfe63658fb3231f128b.